src/mpl/src/shm/mpl_shm_sysv.c

```diff
@@ -44,12 +50,13 @@ static inline int MPL_shm_seg_create_attach_templ(MPL_shm_hnd_t hnd, intptr_t se
     if (flag & MPLI_SHM_FLAG_SHM_ATTACH) {
         /* Attach to shared mem seg */
         *shm_addr_ptr = shmat(MPLI_shm_lhnd_get(hnd), NULL, 0x0);
+        if (*shm_addr_ptr == (void *) -1) {
+            rc = -1;
+        }
     }
 
   fn_exit:
     return rc;
-  fn_fail:
-    goto fn_exit;
 }
```

Comment:
Should this change be in a separate commit ? I.e., remove unused code ?

Reply:
Done